sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -218,10 +218,14 @@ private static void ensureUnionOutputsHaveCompatibleTypes(QueriedRelation left,
             Symbol result = Symbols.tryUpcast(leftSymbol, rightSymbol);
             if (result != null) {
                 leftOutputs.set(i, result);
+                Field oldField = left.fields().get(i);
+                left.fields().set(i, new Field(oldField.relation(), oldField.path(), result.valueType()));
```

Comment:
I wonder, would it make sense to programmatically support this field change? Do we have that in other places?

Reply:
As discussed, we could do:
 - setter for valueType on the Field -> bad option cause fields are immutable up to now
 - setter fields(List<Fields>) on the Analyzed/QueriedRelation -> he haven't done it so far and feels unecessary change for this special case.